adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -580,7 +632,10 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
 
   def loadAlignments(
     filePath: String,
-    projection: Option[Schema] = None): RDD[AlignmentRecord] = LoadAlignmentRecords.time {
+    projection: Option[Schema] = None,
+    filePath2Opt: Option[String] = None,
```

Comment:
What kind of detecting do you think we should do? Inference based on filename extension? I'm not sure I like the sound of that but will defer to you